core/src/main/java/software/amazon/awssdk/core/sync/StreamingResponseHandler.java

```diff
@@ -93,8 +98,34 @@ default boolean needsConnectionLeftOpen() {
      */
     static <ResponseT> StreamingResponseHandler<ResponseT, ResponseT> toFile(Path path) {
         return (resp, in) -> {
-            Files.copy(in, path);
-            return resp;
+            try {
+                Files.copy(in, path);
+                return resp;
+            } catch (IOException copyException) {
+                String copyError = "Failed to read response into file: " + path;
+
+                // If the write failed because of the state of the file, don't retry the request.
+                if (copyException instanceof FileAlreadyExistsException || copyException instanceof DirectoryNotEmptyException) {
+                    throw new IOException(copyError, copyException);
+                }
+
+                // Try to clean up the file so that we can retry the request. If we can't delete it, don't retry the request.
+                try {
+                    Files.deleteIfExists(path);
+                } catch (IOException deletionException) {
+                    Logger.loggerFor(StreamingResponseHandler.class)
```

Comment:
Should we just store this Logger in the class?

Reply:
We can't have private anything in an interface in Java 9.